pkg/cli/login/login.go

```diff
@@ -177,6 +194,31 @@ func (o LoginOptions) Validate(cmd *cobra.Command, serverFlag string, args []str
 		return errors.New("--web cannot be used along with --username, --password or --token")
 	}
 
+	if len(o.OIDCClientID) == 0 && (len(o.OIDCAuthType) > 0 || len(o.OIDCExtraArgs) > 0) {
+		return errors.New("--external-auth-type, --external-extra-args can only be used for external issuers and --external-client-id is required")
+	}
+
+	if len(o.OIDCClientID) > 0 {
+		if len(o.OIDCAuthType) == 0 {
+			return errors.New("auth type should be set and accepted values are oidc and azure")
+		}
+
+		if o.OIDCAuthType == AzureAuthType {
+			found, _ := oidc.ExternalOIDCExtraArgIsSet(o.OIDCExtraArgs, "--tenant-id")
```

Comment:
I agree with you. In any case, it is mandatory for azure kubelogin plugin and I think, we don't need to check this in here because plugin will return error if `tenant-id` is not set.